common/src/main/java/com/wynntils/models/containers/type/InventoryWatcher.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.containers.type;
+
+import java.util.function.Predicate;
+import net.minecraft.world.item.ItemStack;
+
+public class InventoryWatcher {
+    private int slots;
+    private int totalCount;
+    private final Predicate<ItemStack> checker;
+
+    public InventoryWatcher(Predicate<ItemStack> checker) {
+        this.checker = checker;
+    }
+
+    public int getSlots() {
+        return slots;
+    }
+
+    public int getTotalCount() {
+        return totalCount;
+    }
+
+    protected void onUpdate(int oldSlots, int oldTotalCount) {
+        // Called after an update has been made, override if needed
+    }
+
+    // This should only be called by the model
+    public boolean include(ItemStack itemStack) {
```

Comment:
```suggestion
    public boolean includes(ItemStack itemStack) {
```
?

Reply:
I think `includes` just sounded off, like it was something that was included already. I renamed it `shouldInclude`, I hope you think this is okay.

Reply:
If you accept, please resolve, otherwise we can think over the naming a bit more. Maybe it needs to be something more along the lines of it being a predicate...

Reply:
Sounds fine